### PR TITLE
feat: implement ProgressCircular component

### PR DESCRIPTION
**Pull request by alexandprivate**
fix: #45

## Changes proposed in this PR:
feat: implement ProgressCircular component

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/ProgressCircular/ProgressCircularSVG.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+const ProgressCircularSVG: React.FC<Props> = props => {
+    const { variant = 'brand', value = 0 } = props;
+    return (
+        <Svg viewBox="0 0 100 100" style={{ height: '100%', width: '100%' }}>
+            <Path
+                d="M 50,50 m 0,-47 a 47,47 0 1 1 0,94 a 47,47 0 1 1 0,-94"
+                stroke="#ccc"
+                strokeLinecap="round"
+                strokeWidth={1}
```
Comment (TahimiLeonBravo) [inline src/components/ProgressCircular/ProgressCircularSVG.tsx]: We need to fix the `strokeWidth` like the design

File: src/components/ProgressCircular/ProgressCircularSVG.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * A profress circular svg component represents an object or entity
+ */
+
+const ProgressCircularSVG: React.FC<Props> = props => {
+    const { variant = 'brand', value = 0 } = props;
+    return (
+        <Svg viewBox="0 0 100 100" style={{ height: '100%', width: '100%' }}>
+            <Path
+                d="M 50,50 m 0,-47 a 47,47 0 1 1 0,94 a 47,47 0 1 1 0,-94"
+                stroke="#ccc"
```
Comment (TahimiLeonBravo) [inline src/components/ProgressCircular/ProgressCircularSVG.tsx]: We need to use the color `COLOR_GRAY_TRANSPARENT_2`

File: src/components/ProgressCircular/ProgressCircularSVG.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                strokeLinecap="round"
+                strokeWidth={1}
+                fillOpacity={0}
+            />
+            <Path
+                d="M 50,50 m 0,-47 a 47,47 0 1 1 0,94 a 47,47 0 1 1 0,-94"
+                stroke={colors[variant]}
+                strokeLinecap="round"
+                strokeWidth={6}
```
Comment (TahimiLeonBravo) [inline src/components/ProgressCircular/ProgressCircularSVG.tsx]: We need to fix the strokeWidth like the design

File: src/components/ProgressCircular/styled/index.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+interface Props extends BaseProps {
+    size?: Sizes;
+}
+
+const StyledProgressCircular = styled.View<Props>`
+    position: relative;
+    ${props =>
+        props.size === 'large' &&
+        `
+            height: 100;
```
Comment (TahimiLeonBravo) [inline src/components/ProgressCircular/styled/index.ts]: Please use the `width` and `hight` value `86px`

File: src/components/ProgressCircular/styled/index.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            width: 100;
+        `};
+    ${props =>
+        props.size === 'medium' &&
+        `
+            height: 78;
```
Comment (TahimiLeonBravo) [inline src/components/ProgressCircular/styled/index.ts]: Please use the `width` and `hight` value `56px`

File: src/components/ProgressCircular/styled/index.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            width: 78;
+        `};
+    ${props =>
+        props.size === 'small' &&
+        `
+            height: 56;
```
Comment (TahimiLeonBravo) [inline src/components/ProgressCircular/styled/index.ts]: Please use the `width` and `hight` value `40px`

**Comment by TahimiLeonBravo** [comment]
@alexandprivate use this examples for the story
Screen Shot 2019-12-24 at 12 24 03 AM

**Comment by TahimiLeonBravo** [comment]
@alexandprivate please review the design and add this style

Screen Shot 2019-12-24 at 12 25 21 AM

**Comment by alexandprivate** [comment]
> @alexandprivate please review the design and add this style
> 
> Screen Shot 2019-12-24 at 12 25 21 AM

done!

**Comment by alexandprivate** [comment]
> @alexandprivate use this examples for the story
> ProgressCircular

Can you please share those images as SVG to use them as Icons? In Zeplin are formatted as PNG

File: src/components/ProgressCircular/ProgressCircularSVG.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+type Variants = 'brand' | 'success' | 'warning' | 'error';
+
+interface Props extends BaseProps {
+    variant?: Variants;
+    value?: number;
+}
+
+/**
+ * A profress circular svg component represents an object or entity
+ */
```
Comment (LeandroTorresSicilia) [inline src/components/ProgressCircular/ProgressCircularSVG.tsx]: Descriptions only need to be in index.tsx. This descriptions are for future documentation of the component and its props.

File: src/components/ProgressCircular/ProgressCircularSVG.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+const ProgressCircularSVG: React.FC<Props> = props => {
+    const { variant = 'brand', value = 0 } = props;
+    return (
+        <Svg viewBox="0 0 100 100" style={{ height: '100%', width: '100%' }}>
```
Comment (LeandroTorresSicilia) [inline src/components/ProgressCircular/ProgressCircularSVG.tsx]: No use style inlines, instead make a style component for this like:
```export StyledSvg = styled(Svg)``;```

File: src/components/ProgressCircular/ProgressCircularSVG.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+                strokeWidth={1}
+                fillOpacity={0}
+            />
+            <Path
+                d="M 50,50 m 0,-47 a 47,47 0 1 1 0,94 a 47,47 0 1 1 0,-94"
+                stroke={colors[variant]}
+                strokeLinecap="round"
+                strokeWidth={6}
+                fillOpacity={0}
+                strokeDasharray={`${(value / 100) * 300} 300`}
```
Comment (LeandroTorresSicilia) [inline src/components/ProgressCircular/ProgressCircularSVG.tsx]: Let's not mix js in jsx in order to do better separation of concerns and put this in a const before the return statement.

File: src/components/ProgressCircular/ProgressCircularSVG.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+            />
+        </Svg>
+    );
+};
+
+ProgressCircularSVG.propTypes = {
+    /**Variant defines the colors of the progress circular it could be one of 'brand' | 'success' | 'warning' | 'error'
+     * default value is brand
+     */
```
Comment (LeandroTorresSicilia) [inline src/components/ProgressCircular/ProgressCircularSVG.tsx]: same ^^ descriptions only are needed in index

File: src/components/ProgressCircular/index.tsx [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    StyledProgressCircular,
+    StyledIconContainer,
+    StyledIcon,
+} from './styled';
+import RenderIf from '../RenderIf';
+import ProgressCircularSVG from './ProgressCircularSVG';
+import { BaseProps } from '../types';
+
+type Sizes = 'large' | 'medium' | 'small';
+type Variants = 'brand' | 'success' | 'warning' | 'error';
```
Comment (LeandroTorresSicilia) [inline src/components/ProgressCircular/index.tsx]: Let's define this in only one place. You can export this here and not redeclare it in
```ProgressCircularSVG.tsx```

File: src/components/ProgressCircular/index.tsx [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+interface Props extends BaseProps {
+    variant?: Variants;
+    icon?: ReactNode;
+    size?: Sizes;
+    value?: number;
+}
+
+/**
+ * A profress circular component represents an object or entity
```
Comment (LeandroTorresSicilia) [inline src/components/ProgressCircular/index.tsx]: Follow the description in the API documentation:
```ProgressCircular component communicates to the user the progress of a particular process.```

File: src/components/ProgressCircular/index.tsx [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+    variant: PropTypes.oneOf(['brand', 'success', 'warning', 'error']),
+    /** The the icon centered at the progress circular.
+     * default value is null
+     */
+    icon: PropTypes.node,
+    /** The size of the progress circular. One of small, medium, and large.
+     * This value defaults to medium. */
+    size: PropTypes.oneOf(['large', 'medium', 'small']),
+    /**the value percentage of the progress circular
+     * default 0
```
Comment (LeandroTorresSicilia) [inline src/components/ProgressCircular/index.tsx]: The percentage value of the progress circular. It must be between 0 and 100.
Also start with capital case after a ```.```. Review all descriptions.

File: src/components/ProgressCircular/styled/index.ts [hunk 14]
```diff
@@ -0,0 +1,49 @@
+import styled from 'styled-components/native';
+import { BaseProps } from '../../types';
+
+type Sizes = 'large' | 'medium' | 'small';
```
Comment (LeandroTorresSicilia) [inline src/components/ProgressCircular/styled/index.ts]: same ^^ define in one place

File: src/components/ProgressCircular/styled/index.ts [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+    align-items: center;
+    justify-content: center;
+    top: 0;
+    left: 0;
+    height: 100%;
+    width: 100%;
+`;
+
+const StyledIcon = styled.View<{}>`
+    /* position: relative; */
```
Comment (LeandroTorresSicilia) [inline src/components/ProgressCircular/styled/index.ts]: Clean comment codes.

File: src/components/ProgressCircular/ProgressCircularSVG.tsx (hunk 8 above)
Comment (alexandprivate) [reply src/components/ProgressCircular/ProgressCircularSVG.tsx]: I change that already

**Pull request merged**